pkg/registry/garden/backupinfrastructure/strategy.go

```diff
@@ -75,6 +75,10 @@ func mustIncreaseGeneration(oldBackupInfrastructure, newBackupInfrastructure *ga
 		return true
 	}
 
+	if val, ok := newBackupInfrastructure.Annotations[common.BackupInfrastructureOperation]; ok && val == common.BackupInfrastructureReconcile {
```

Comment:
Can you maybe add a `utils/kubernetes` function like (best next to `SetMetaDataLabel`) called `HasMetaDataAnnotation` that checks if a given value is present and set to the given value? This way we could also reuse code / refactor this a bit.